src/ng-select/ng-select.component.ts

```diff
@@ -179,12 +182,13 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     }
 
     ngOnDestroy() {
-        this.changeDetectorRef.detach();
```

Comment:
Check if it works when using *ngIf on ng-select (example is in reactive forms)

Reply:
@anjmao it works